src/rust/engine/process_execution/src/local.rs

```diff
@@ -135,8 +149,16 @@ impl super::CommandRunner for CommandRunner {
             )
           })
           // Force workdir not to get dropped until after we've ingested the outputs
-          .map(|result| (result, workdir))
-          .map(|(result, _workdir)| result)
+          .map(move |result| (result, workdir, cleanup_local_dirs) )
+          .map(move |(result, workdir, cleanup_local_dirs)| {
+            if cleanup_local_dirs == false {
+              // This consumes the `TempDir` without deleting directory on the filesystem, meaning
+              // that the temporary directory will no longer be automatically deleted when dropped.
+              let preserved_path = workdir.into_path();
+              info!("preserved local process execution dir: {:?}", preserved_path.into_os_string());
```

Comment:
May also want to include the execution request (or at least the description field from it) in this log, to help people know which directory they're interested in

Reply:
done